opentaskspal/src/main/java/org/dmfs/opentaskspal/tasks/RelationData.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasks;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.RowSnapshot;
+import org.dmfs.android.contentpal.rowdata.Composite;
+import org.dmfs.android.contentpal.rowdata.DelegatingRowData;
+import org.dmfs.android.contentpal.rowdata.RawRowData;
+import org.dmfs.android.contentpal.rowdata.Referring;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * {@link RowData} for adding a {@link TaskContract.Property.Relation} property.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class RelationData extends DelegatingRowData<TaskContract.Properties>
+{
```

Comment:
For some use cases (i.e. syncing) and also to test the provider we also need a ctor which takes a task UID (in form of a `CharSequence`) instead of a `RowSnapshot`.

Reply:
Just to confirm before I do anything, does it mean that we need to look up the `RowSnapshot` with something like this?:
```
new QueryRowSet(view, new HasUid(uid)).iterator().next()
```
Or, since sync uid is a unique id as well, would maybe the provider support looking the row up with url? Like with the standard id.

Reply:
You don't need to look it up. It's part of the contract that you can either pass an id, a uid or a uri : https://github.com/dmfs/opentasks/blob/master/opentasks-contract/src/main/java/org/dmfs/tasks/contract/TaskContract.java#L1494
The provider will look up the task and insert the id if none is given (see https://github.com/dmfs/opentasks/blob/master/opentasks-provider/src/main/java/org/dmfs/provider/tasks/handler/RelationHandler.java#L115)
Note that the UID can belong to a task which has not been synced yet or even to another calendar object (like and event or journal entry).

Reply:
Added the extra constructor.